build-logic/settings.gradle.kts

```diff
@@ -4,3 +4,54 @@ include("kotlin-convention")
 include("android-convention")
 include("testing-convention")
 include("publication")
+
+val artifactoryUrl: String? by settings
+
+fun MavenArtifactRepository.artifactoryUrl(repositoryName: String) {
+    setUrl("$artifactoryUrl/$repositoryName")
+    isAllowInsecureProtocol = true
+}
+
+fun MavenArtifactRepository.setUrlOrProxy(repositoryName: String, originalRepo: String) {
+    if (artifactoryUrl.isNullOrBlank()) {
+        name = repositoryName
+        setUrl(originalRepo)
+    } else {
+        name = "Proxy for $repositoryName: $originalRepo"
+        artifactoryUrl(repositoryName)
+    }
+}
+
+dependencyResolutionManagement {
```

Comment:
Don't we need the same for `pluginManagement.repositories`?

Reply:
yes, you're right